### ISSUE TITLE
Doesn't seem to work behind a corporate proxy?

### ISSUE THREAD

**Issue by teehemkay** [felixrieseberg/npm-windows-upgrade#24]
I followed the installation instructions on the README.
Then, when I try to run the tool, I get an error:

```
λ npm-windows-upgrade                                                                       
npm-windows-upgrade 0.3.2                                                                   
? This tool will upgrade npm. Do you want to continue? Yes                                  
R:\java\tools\npm\node_modules\npm-windows-upgrade\bin\npm-windows-upgrade:294              
                return console.err(chalk.bold.red('We have trouble connecting               
                               ^                                                            
TypeError: undefined is not a function                                                      
    at R:\java\tools\npm\node_modules\npm-windows-upgrade\bin\npm-windows-upgrade:294:32    
    at R:\java\tools\npm\node_modules\npm-windows-upgrade\bin\npm-windows-upgrade:55:13     
    at GetAddrInfoReqWrap.asyncCallback [as callback] (dns.js:81:16)                        
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:94:17)                           

```

There seems to be two problems.

1. Re: the "We have trouble connecting to the Internet..." error message seems to indicate that npm-windows-upgrade seems to have trouble connecting to the Internet and I'm not sure why.
I'm in a corporate environment behind a proxy but npm is configured correctly to use the proxy and it has no problem doing downloading packages from npmjs.org example. 
Maybe this is happening because I'm behind a  corporate proxy and the script does not correctly pick up the proxy settings like npm does?

2. npm-windows-upgrade then crashes trying to report the above issue.

I'm using node v0.12.7 and npm 2.11.3. behing a proxy

**Comment by felixrieseberg** [comment]
Hey, thanks for reporting! I didn't really consider corporate proxies, so this one is on me - I'll add an option to override the usual Internet check (which is just a DNS ping to microsoft.com).

**Comment by felixrieseberg** [comment]
Thanks again! This feature is now in 0.4.0 (as `--no-dns-check`), which was just published :+1:

**Comment by teehemkay** [comment]
Fantastic!

Thanks for fixing this so quickly.
I'll test the fix on Monday.

**Comment by teehemkay** [comment]
I can confirm that both issues are corrected in 4.0 :+1:.
- Error message correctly displayed in case of Internet connection failure
- `--no-dns-check`option works as advertise.

Thanks again for the enhancement!
